### DISCUSSION TITLE
Why switch to GPL

### DISCUSSION THREAD

**Discussion by fakerybakery** [GaiZhenbiao/ChuanhuChatGPT#957] [General]
Why switch from MIT -> GPL

**Comment by GaiZhenbiao** [comment]
Because Linus Torvalds is cool :)

**Comment by fakerybakery** [comment]
Hi, I thought Linus Torvalds was [vehemently](https://youtu.be/PaKIZ7gJlRU?t=240) [anti-GPLv3](https://www.youtube.com/watch?v=PaKIZ7gJlRU)?

**Comment by Keldos-Li** [comment]
@fakerybakery Although I don't appreciate your tone, this is indeed an interesting question. After revisiting some relevant information, I've decided to respond.

Initially, the license at the project's creation doesn't necessarily reflect a carefully considered choice. We switched to GPL3 before the project gained traction, and you can disregard the previous license, as it went unnoticed back then.

Regarding Linus, his preference for GPL2 over GPL3 is rooted in disagreements about cracking hardware encryption. Our software doesn't run on hardware kernels, eliminating such concerns. Linus supports GPL due to the core principle: "I give you the source code, you give me your modifications." This mandatory feedback is crucial for continuous software improvement. GPL2 and GPL3 don't differ significantly in this regard, but GPL3 provides better legal protection for authors. That's why many prefer GPL3.

If you're wondering why we don't use more permissive licenses like MIT, that's because our program isn't fundamentally a low-level framework. It's designed for direct user interaction, and we take pride in our product. We don't want it casually modified, redistributed, losing our copyright, or someone using a modified product without knowing its origin.

**Reply by fakerybakery** [reply]
Hi, thanks for your response. I deeply apologize if my tone seemed offensive or adversarial, as this was not my intention. I understand your choice to switch to GPL and would like to thank you for creating this project.

**Reply by Keldos-Li** [reply]
I've penned some notes regarding licensing matters for our product [here](https://github.com/GaiZhenbiao/ChuanhuChatGPT/wiki/使用许可). I trust this will comprehensively address your concerns. However, I am curious about why you "decided to stop using GPL" 👀.
